### PR TITLE
Retrieve the fractions associated with a given aperture/photometry

### PR DESCRIPTION
**Pull request by fdeugenio**
At the moment the methods do_photometry & co. accept only 2d input data.

This is problematic when doing aperture photometry on a data cube (e.g. create aperture spectra).
The fastest method is to get the pixel weights (fractions) directly.

An alternative implementation would see heavier edits, with an added ``axes`` keyword which specifies which axes will be suppressed by ``do_photometry``.

### CODE REVIEW

File: photutils/aperture_core.py [hunk 1]
```diff
@@ -186,7 +187,47 @@ def do_photometry(self, data, error=None, effective_gain=None,
...
+        subpixels : int, optional
+            For the ``'subpixel'`` method, resample pixels by this factor (in
+            each dimension). That is, each pixel is divided into
+            ``subpixels ** 2`` subpixels.
+
+        Returns
+        -------
+        fraction : `numpy.array`
+            Array with the same shape as ``data``. Each element is the fraction
+        of the corresponding ``data`` pixel that falls within the aperture.
```
Comment (larrybradley) [inline photutils/aperture_core.py]: This line should be indented to match the line above.

**Comment by larrybradley** [comment]
Thanks, @fdeugenio!  This looks good.  Could you please add some tests for the new methods?

**Comment by larrybradley** [comment]
And it would be great to perhaps include a simple test for your use case of performing aperture photometry on a (small) data cube.

**Comment by fdeugenio** [comment]
dear @larrybradley , I can certainly add some tests - but the timescale for this would be about 1.5 months. Is that still ok?

**Comment by larrybradley** [comment]
In that case I'll probably merge this soon.  You can add the tests later if they haven't already been added.  Thanks.

**Comment by fdeugenio** [comment]
Sorry Larry,

I am not allowed can't take my laptop on holidays. I'll see what can I 
do from my phone though :-(

**Comment by larrybradley** [comment]
No worries.  Thanks for your contribution, @fdeugenio.

**Pull request merged**